### PR TITLE
connect: Add Envoy 1.22 to integration tests, remove Envoy 1.18

### PR DESCRIPTION
**Pull request by eculver**
Envoy 1.22.0 was released today. This adds it to our test matrix and removes 1.18.6.

### CODE REVIEW

File: agent/xds/envoy_versioning.go [hunk 1]
```diff
@@ -11,7 +11,7 @@ import (
 var (
 	// minSupportedVersion is the oldest mainline version we support. This should always be
 	// the zero'th point release of the last element of proxysupport.EnvoyVersions.
-	minSupportedVersion = version.Must(version.NewVersion("1.18.0"))
+	minSupportedVersion = version.Must(version.NewVersion("1.19.0"))
 
 	minVersionToForceLDSandCDSToAlwaysUseWildcardsOnReconnect = version.Must(version.NewVersion("1.19.0"))
```
Comment (rboyer) [inline agent/xds/envoy_versioning.go:16]: This can be deleted too now.

**Review by rboyer — APPROVED** [review]
LGTM assuming tests pass and you remove that one last piece of dead code.

File: agent/xds/envoy_versioning.go (hunk 1 above)
Comment (eculver) [reply agent/xds/envoy_versioning.go:16]: ahh right, I meant to get that. thanks!

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/642386.

**Comment by hc-github-team-consul-core** [comment]
:cherries::x: Cherry pick of commit 000d0621b453b727ed9236156251a79ddb981d85 onto `stable-website` failed! [Build Log](https://circleci.com/gh/hashicorp/consul/642386)

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit 000d0621b453b727ed9236156251a79ddb981d85 onto `release/1.12.x` succeeded!

**Comment by dekimsey** [comment]
@eculver just drawing attention to this. Consul 1.12 with Envoy is effectively no longer supported on CentOS 7 with this change. The [packaged envoy](https://www.envoyproxy.io/docs/envoy/latest/start/install#install-envoy-on-rpm-based-distros) was getenvoy-envoy from tetratelabs, and builds [stopped working](https://github.com/tetratelabs/archive-envoy/issues/17) with 1.19. CentOS 7 packages have been stuck on 1.18 for sometime while waiting for official builds (if they ever happen).

**Pull request merged**